src/providers/ogr/qgsogrfeatureiterator.cpp

```diff
@@ -45,6 +45,8 @@ QgsOgrFeatureIterator::QgsOgrFeatureIterator( QgsOgrFeatureSource *source, bool
   , mFilterFidsIt( mFilterFids.constBegin() )
   , mSharedDS( source->mSharedDS )
 {
+  // Since connection timeout for OGR connections is problematic and can lead to crashes, disable for now.
```

Comment:
No idea why to remove timeout for OGR, but if it is planned to work on later, probably is better to comment with "TODO:"

Reply:
@luipir , OGR timeout == crash at the moment. It's ideally a TODO: that'll be picked up by someone with proper knowledge on our connection pooling code.